add Watson Ladd's suggested text with minor adaptations (#535)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

hours and calculate `exp` accordingly) or rounded (e.g., rounded down to the
beginning of the day).

> SD-JWT only conceals the value of claims that aren't revealed.

Reviewer: ```suggestion
SD-JWT only conceals the value of claims that are not revealed.
```

> It does not meet standard security notations for anonymous credentials. In

Reviewer 2: Can we have a reference to what "anonymous credentials"? do we really expect all of the readers to know what it is? this is the first time that they are mentioned in the text I believe?

Reviewer 3: I think the idea was to add a citation to CL01, but that might not actually have the phrase in it easily findable...

Reviewer: I added the citation below.

particular Verifiers and Issuers can know when they have seen the same

> credential no matter what fields have been disclosed, even none of them.

Reviewer 2: ```suggestion
credential no matter what fields have been disclosed, even when none .
```

Reviewer: ```suggestion
credential no matter what fields have been disclosed, even when none are disclosed.
```

> This behavior may not accord with what users intuitively expect or are
> lead to expect from user interface interactions and lead to them make choices they
> would not otherwise make. Workarounds such as batch issuance as

Reviewer: ```suggestion
This behavior may not align with what users naturally anticipate or are guided to
expect from user interface interactions, potentially causing them to make decisions
they might not otherwise make. Workarounds such as batch issuance as
```

> described above help for keeping
> Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.

Reviewer 2: ```suggestion
described above help with keeping
Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.
```

> would not otherwise make. Workarounds such as issuing multiple
> credentials at once and using them only one time can help for keeping
> Verifiers from linking different presentations, but cannot work for Issuers.

Reviewer: ```suggestion
would not otherwise make. Workarounds such as batch issuance as
described above help for keeping
Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.
```

> SD-JWT only conceals the value of claims that are not revealed.
> It does not meet standard security notations for anonymous credentials. In
> particular Verifiers and Issuers can know when they have seen the same
> credential no matter what fields have been disclosed, even none of them.
> This behavior may not accord with what users intuitively expect or are
> lead to expect from user interface interactions and lead to them make choices they
> would not otherwise make. Workarounds such as batch issuance as
> described above help for keeping
> Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.
> This issue applies to all salted hash-based approaches,
> including mDL/mDoc [@?ISO.18013-5] and SD-CWT [@?I-D.ietf-spice-sd-cwt].
> 

Reviewer: ```suggestion
SD-JWT only conceals the value of claims that are not revealed.
It does not provide Issuer/Verifier unlinkability without constraints
as schemes using zero-knowledge protocols do (for example, [@?CL01]). In
particular Verifiers and Issuers can know when they have seen the same
credential no matter what fields have been disclosed, even none of them.
This behavior may not accord with what users intuitively expect or are
lead to expect from user interface interactions and lead to them make choices they
would not otherwise make. Workarounds such as batch issuance as
described above help for keeping
Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.
This issue applies to all salted hash-based approaches,
including mDL/mDoc [@?ISO.18013-5] and SD-CWT [@?I-D.ietf-spice-sd-cwt].

```

Author: is `[@?CL01]` just gonna work? I need to reevaluate some of my belief systems, if it does.

Author: do not schemes using zero-knowledge protocols also have constraints with respect to Issuer/Verifier unlinkability? Just somewhat less constraining constraints.

Reviewer 2: please combine this suggestion with the suggestions I have made above :)

Reviewer 3: @bc-pi Exactly. Saying without further conditions doesn't quite work, saying what the exact notion is is technical, pointing to a reference is the happy medium and lets us get the important part across.

> This issue applies to all selective disclosure based approaches,
> including mDL and mDoc [@?ISO.18013-5], and SD-CWT [@?I-D.ietf-spice-sd-cwt].

Reviewer: ```suggestion
This issue applies to all salted hash-based approaches,
including mDL/mDoc [@?ISO.18013-5] and SD-CWT [@?I-D.ietf-spice-sd-cwt].
```

> SD-JWT only conceals the value of claims that are not revealed.
> It does not meet security notations for anonymous credentials [@?CL01]. In
> particular, colluding Verifiers and Issuers can know when they have seen the same
> credential no matter what fields have been disclosed, even when none are disclosed.
> This behavior may not align with what users naturally anticipate or are guided to
> expect from user interface interactions, potentially causing them to make decisions
> they might not otherwise make. Workarounds such as batch issuance as
> described above help with keeping
> Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.
> This issue applies to all salted hash-based approaches,
> including mDL/mDoc [@?ISO.18013-5] and SD-CWT [@?I-D.ietf-spice-sd-cwt].

Reviewer 2: ```suggestion
SD-JWT only conceals the value of claims that are not revealed.
It does not meet security notations for anonymous credentials [@?CL01]. In
particular, colluding Verifiers and Issuers can know when they have seen the same
credential no matter what fields have been disclosed, even when none have been disclosed.
This behavior may not align with what users naturally anticipate or are guided to
expect from user interface interactions, potentially causing them to make decisions
they might not otherwise make. Workarounds such as batch issuance, as
described above, help with keeping
Verifiers from linking different presentations, but cannot work for Issuer/Verifier unlinkability.
This issue applies to all salted hash-based approaches,
including mDL/mDoc [@?ISO.18013-5] and SD-CWT [@?I-D.ietf-spice-sd-cwt].
```

 <front>
  <title>JSON Web Token Claims</title>
  <author><organization>IANA</organization></author>
</front>
</reference>

<reference anchor="ISO.18013-5" target="https://www.iso.org/standard/69084.html">
<front>

>   <title>ISO/IEC 18013-5:2021 Personal identification — ISO-compliant driving license — Part 5: Mobile driving license (mDL)  application</title>

Author: ```suggestion
  <title>ISO/IEC 18013-5:2021 Personal identification — ISO-compliant driving license — Part 5: Mobile driving license (mDL) application</title>
```

 <front>
  <title>JSON Web Token Claims</title>
  <author><organization>IANA</organization></author>
</front>
</reference>

> <reference anchor="ISO.18013-5" target="https://www.iso.org/standard/69084.html">
> <front>
>   <title>ISO/IEC 18013-5:2021 Personal identification — ISO-compliant driving license — Part 5: Mobile driving license (mDL) application</title>
>   <author>
>     <organization> ISO/IEC JTC 1/SC 17 Cards and security devices for personal identification</organization>
>   </author>
>   <date year="2021"/>
> </front>
> <reference anchor="CL01" target="https://eprint.iacr.org/2001/019.pdf">
>   <front>
>     <title>An Efficient System for Non-Transferable Anonymous Credentials with Optional Anonymity Revocation</title>
>     <author initials="J." surname="Camenisch" fullname="Jan Camenisch">
>       <organization>IBM Research</organization>
>     </author>
>     <author initials="A." surname="Lysyanskaya" fullname="Anna Lysyanskaya">
>       <organization>MIT</organization>
>     </author>
>     <date year="2001" />
>   </front>
>   <seriesInfo name="Proceedings of the International Conference on the Theory and Application of Cryptographic Techniques (EUROCRYPT)" value="2001" />
> </reference>
> </reference>

Reviewer 2: ```suggestion
<reference anchor="ISO.18013-5" target="https://www.iso.org/standard/69084.html">
<front>
  <title>ISO/IEC 18013-5:2021 Personal identification — ISO-compliant driving license — Part 5: Mobile driving license (mDL) application</title>
  <author>
    <organization> ISO/IEC JTC 1/SC 17 Cards and security devices for personal identification</organization>
  </author>
  <date year="2021"/>
</front>
</reference>

<reference anchor="CL01" target="https://eprint.iacr.org/2001/019.pdf">
  <front>
    <title>An Efficient System for Non-Transferable Anonymous Credentials with Optional Anonymity Revocation</title>
    <author initials="J." surname="Camenisch" fullname="Jan Camenisch">
      <organization>IBM Research</organization>
    </author>
    <author initials="A." surname="Lysyanskaya" fullname="Anna Lysyanskaya">
      <organization>MIT</organization>
    </author>
    <date year="2001" />
  </front>
  <seriesInfo name="Proceedings of the International Conference on the Theory and Application of Cryptographic Techniques (EUROCRYPT)" value="2001" />
</reference>
```
